modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -67,7 +73,7 @@ def execute cmd, msg=nil, error_only=false
         std_err.close unless std_in.nil?
       end
       report msg, response, error_only
-      response
+      response if return_result
```

Comment:
nowhere in the code where it's used the result is needed. No need for conditionals (and additional tests), just don't return anything from the method.